uvicorn/config.py

```diff
@@ -215,10 +250,15 @@ def __reduce__(self):
             logging.Logger.__reduce__ = __reduce__
 
         if self.log_config is not None:
+            if not self.use_colors:
+                self.log_config = LOGGING_CONFIG_NO_COLORS
```

Comment:
Instead of switching the entire config here, lets use...

```python
if self.use_colors in (True, False):
    self.log_config["formatters"]["default"]["use_colors"] = self.use_colors
```

We should only do this within the `if isinstance(self.log_config, dict):` case. If someone is using the file config then it's up to them to choose the correct settings there.

Reply:
added the same logic for access too